caddytest/caddytest.go

```diff
@@ -371,3 +392,75 @@ func AssertAdapt(t *testing.T, rawConfig string, adapterName string, expectedRes
 		t.Fail()
 	}
 }
+
+// AssertPostResponse asserts whether POST request returns expected status code and body.
+func AssertPostResponse(t *testing.T, requestURI string, requestHeaders []string, requestBody *bytes.Buffer, statusCode int, expectedBody string) (*http.Response, string) {
+	tester := NewTester(t)
+	return tester.AssertPostResponse(requestURI, requestHeaders, requestBody, statusCode, expectedBody)
+}
+
+// AssertPostResponse asserts whether POST request returns expected status code and body.
+func (tc *Tester) AssertPostResponse(requestURI string, requestHeaders []string, requestBody *bytes.Buffer, statusCode int, expectedBody string) (*http.Response, string) {
+	resp, body := AssertPostResponseBody(tc, requestURI, requestHeaders, requestBody, statusCode)
+	if !strings.Contains(body, expectedBody) {
```

Comment:
You had an observation about assertions with html that was stripped of whitespaces. If there is a tester object perhaps we could deal with this by putting a transformation on the body.

The null implementation would be
`tc.TransformBody := func(resp *http.Response, body string) string { return body }`
This would allow for testing code bases to figure out how to transform the response to make the assertion cleaner.

Reply:
> You had an observation about assertions with html that was stripped of whitespaces. If there is a tester object perhaps we could deal with this by putting a transformation on the body.

@sarge , it looks like it got fixed by someone. I don't have that problem now.

Reply:
OK we can leave this out for now.